src/node/catchup.ml

```
@@ -297,9 +320,28 @@ let verify_n_catchup_store (type s) ~stop me ?(apply_last_tlog_value=false) ((mo
     (Sn.string_of too_far_i) (Sn.string_of current_i) (io_s si_o) >>= fun () ->
    match too_far_i, si_o with
     | i, None when i <= 0L -> Lwt.return ()
-    | i, Some j when i = j -> Lwt.return ()
+    | i, Some j when i = j ->
+      begin
+        let store_cs = S.get_checksum store in
+        let tlog_cs = tlog_coll # get_previous_checksum (Sn.succ i) in
+        if store_cs <> tlog_cs
+        then Lwt.fail (StoreChecksumError (i, store_cs, tlog_cs))
+        else Lwt.return ()
+      end
     | i, Some j when i > j ->
-      catchup_store ~stop me ((module S),store,tlog_coll) too_far_i
+      let entry = ref None in
+      let check e = Lwt.return (entry := Some e) in
+      tlog_coll # iterate j (Sn.succ j) check >>= fun () ->
+      let store_cs = S.get_checksum store in
+      let tlog_cs =
+        match !entry with
+          | None -> None
```

this case should never happen, right?
so you may want to throw an exception